[wasm] Automatically set EMSDK_PATH when a local installation of emsc…  …ripten exists.

vargaz opened the pull request.

vargaz committed:
[wasm] Automatically set EMSDK_PATH when a local installation of emscripten exists.

kg commented on src/mono/wasm/wasm.targets:
```diff
@@ -24,7 +24,6 @@
   <Target Name="CheckEnv">
     <Error Condition="'$(TargetArchitecture)' != 'wasm'" Text="Expected TargetArchitecture==wasm, got '$(TargetArchitecture)'"/>
     <Error Condition="'$(TargetOS)' != 'Browser'" Text="Expected TargetOS==Browser, got '$(TargetOS)'"/>
-    <Error Condition="'$(EMSDK_PATH)' == ''" Text="The EMSDK_PATH environment variable should be set pointing to the emscripten SDK root dir."/>
```
I think we should consider keeping this and having a way to disable the automatic provisioning, maybe with a separate environment variable

vargaz replied:
Its set automatically to src/mono/wasm/emsdk.

vargaz commented:
Will make this non-automatic.

vargaz commented:
Rewrote so we no longer auto download the sdk, just suggest how to do it. Also moved the checks to a separate target so its executed for the build step as well, not just for the configure step.

akoeplinger commented on src/mono/mono.proj:
```diff
@@ -39,8 +39,17 @@
+
+    <PropertyGroup Condition="'$(TargetsBrowser)' == 'true'">
+      <EMSDK_PATH Condition="'$(EMSDK_PATH)' == ''">$(MonoProjectRoot)wasm/emsdk</EMSDK_PATH>
+    </PropertyGroup>
```
This won't work as is since we can't propagate the EMSDK_PATH value from here to the libraries build, so it would only work for the mono runtime build.

In my opinion what we should do instead is print out a message if EMSDK_PATH isn't set telling the user to either set it, or run the `provision-wasm` script which would download emscripten and print out `EMSDK_PATH=.../src/mono/wasm/emsdk` which the user can the copy and rerun the build.

akoeplinger commented on src/mono/wasm/Makefile:
```diff
@@ -42,6 +42,9 @@ else
 endif
 
+check-provision-wasm:
+	@if [ ! -d emsdk ]; then echo "Either set EMSDK_PATH to the path of the emscripten SDK, or install a local copy by running make provision-wasm in src/mono/wasm."; exit 1; fi
```
but like I said above I think we should change this so we print the EMSDK_PATH in provision-wasm and keep the existence check in msbuild.

vargaz committed:
Don't automatically set EMSDK_PATH.

vargaz commented:
Will rework this.
